amqp-replay.js

```diff
@@ -43,13 +45,39 @@ require('amqplib').connect(config.amqp.uri).then(function(conn) {
         if(exchangeCh.publish(config.amqp.exchange.name, routingKey, content, properties)){
           queueCh.ack(msg);
         }
-      }, {
-        noAck: config.amqp.noAck
-      });
+      }
+
+      function getAllMessagesInQueue(res = []) {
+        return queueCh.get(config.amqp.queue.name, {noAck: config.amqp.noAck})
+          .then(newVal => {
+            if (newVal === false) {
+              return res;
+            }
+
+            res.push(newVal);
+            return getAllMessagesInQueue(res);
+          });
+      }
+
+      if (config.infinite) {
+        //replay all messages in queue, infinitely
+        return queueCh.consume(config.amqp.queue.name, handleMessage, {noAck: config.amqp.noAck})
+          .then(_consumeOk => logger.debug('Waiting for messages. To exit press CTRL+C'))
+      } else {
+        //Get all messages currently in queue, and exit once they are replay
+        return getAllMessagesInQueue().then(messages => {
```

Comment:
quid when the queue is about 1 000 000 messages ? We should have a default batch value (customizable through env. variables)